### DISCUSSION TITLE
Migrate to React (TS) Issue #22

### DISCUSSION THREAD

**Discussion by AbhinavReddy-Dev** [ChicoState/PantryNode#30] [Ideas]
Wanted to know if we are going to create a separate folder for the frontend to make progress with migration.

**Comment by AbhinavReddy-Dev** [comment]
Here are some of the resources for TailwindCSS, if we choose this as our interface styling framework.
- https://www.youtube.com/@TailwindLabs/videos
- https://tailwindcss.com/docs/editor-setup
- https://www.hyperui.dev/ , we can use this as a style guide to have uniform UI styling through out the website and also accelerate our process to develop the UI.

**Comment by AbhinavReddy-Dev** [comment]
Please share resources for material UI style guide.

**Reply by AbhinavReddy-Dev** [reply]
> Please share resources for material UI style guide.

https://github.com/ChicoState/PantryNode/issues/46#issuecomment-1457184253

**Comment by briswells** [comment]
I was just looking at the charts page and it looks like it's doing a good bit of very inefficient math that I feel should be handled by the backend. I could see how this would get implemented since they were using a document based database but with a relational I don't see the point. Can we migrate this functionality in parallel with the front end rewrite?

**Reply by AbhinavReddy-Dev** [reply]
Would really appreciate it, its also a bad practice to have those business/data calculations on the frontend.

**Comment by AbhinavReddy-Dev** [comment]
After going through the pages, I realized that we have pages that can just be a simple modal component in the whole page, like adding stock from the stock page, in the current design it takes the user to a new a page to add stock. My proposal is to change this to a modal component to pop-up when the user wants to add stock from the stock list page and also the notification modal component to the user about the successful registration instead of a seperate page. Please add to this discussion.

**Comment by AbhinavReddy-Dev** [comment]
I have added a navigation bar that's common with the admin type dashboards, there is a pages folder where everyone can work on the pages and those pages will be on the right side of the drawer-nav bar.

<img width="828" alt="Screenshot 2023-03-25 at 10 42 15 PM" src="https://user-images.githubusercontent.com/33514341/227757596-11c742d2-660f-4309-9d59-ac665628e9d4.png">

**Comment by hardikpatil** [comment]
I was going through pages, those images on the food categories on the sale page would be static or dynamic? In the original codebase found those images in assets. In our current implementation, category images would receive from the backend Or going to be static? I believe category names would be generated from backend but what about images? @briswells

**Reply by briswells** [reply]
That's on of the nonsensical things about the original implementation. It allowed you to create categories but those images were static assets, thus the reports page didn't reflect any changes in categories. I'm thinking for now we keep them as static assets or just remove them entirely, but we can add an issue for handling them dynamically. This will require the front end team to update the add category form to accept images, and us to store those images in a database. After we've done that we'll need to update backend routes to supply the images, and frontend pages will need to display them. I'll create a backlog issue that we can start working on once we have site in a more stable state.

**Reply by hardikpatil** [reply]
For now, yeah was thinking to keep it both as a static category name and its icons #78 . Later, can create backlog issue for handling them dynamically in the future #82

**Reply by AbhinavReddy-Dev** [reply]
The categories are not many and the situation of adding a category is rare but when we white label this application, admins would like to have an option of adding and removing a category (will have cascading effect throughout the tables) as necessary. My suggestion is to have a table for the categories and have 3-4 keys (id, name, image _etc_).

**Reply by hardikpatil** [reply]
> The categories are not many and the situation of adding a category is rare but when we white label this application, admins would like to have an option of adding and removing a category (will have cascading effect throughout the tables) as necessary. My suggestion is to have a table for the categories and have 3-4 keys (id, name, image _etc_).

Agreed. Last time I checked migrated db schema, there's already Category table/model however, not sure, if there's category icon/image field.

**Reply by briswells** [reply]
@AbhinavReddy-Dev @hardikpatil added my thought about this to #82 I think we should move all discussion over there to avoid cluttering this issue.

**Comment by AbhinavReddy-Dev** [comment]
A request to everyone working on the frontend pages/ components, please have a discussion with the backend team regarding the api and the responses as the content depends on that. This will help reduce the number of times you have to make changes once a commit is done wrt a page functionality.

**Reply by briswells** [reply]
Just as a follow up on that. We've been trying to keep everything consistent with the old data format, except for the charts page which has been discussed already. If for whatever reason you want more or different data just let us know.

**Reply by AbhinavReddy-Dev** [reply]
Sure, because it was written a couple years ago, we can think of something that is relevant today for us in terms of features or data, hoping that we have the flexibility to add more data fields to the apis, or even create new ones.

**Reply by briswells** [reply]
Take a looksee at my schema, I'm all about adding tons of new stuff, but with some many teams all trying to move forward with different things we wanted to keep development from diverging.

**Reply by hardikpatil** [reply]
Thats great, really appreciated. Do you mind, sharing the source/directory path, again, to look up the schema?

**Reply by briswells** [reply]
> Thats great, really appreciated. Do you mind, sharing the source/directory path, again, to look up the schema?

The most up to date schema is in main docs/[Foodbank ER Diagram.png](https://github.com/ChicoState/PantryNode/blob/main/docs/Foodbank%20ER%20Diagram.png). Of course this isn't set in stone, just sometime I threw together as a baseline.

**Comment by AbhinavReddy-Dev** [comment]
@prestonmasseyblake Can you please check the implementation of `frontend_experimental_nav` branch for responsive nav bar? Let me know if it is alright, then I will help someone implement it in the `frontend` branch to be merged into `main`.

**Comment by prestonmasseyblake** [comment]
ok I will take a look

**Comment by AbhinavReddy-Dev** [comment]
Please join this discord server to communicate and pair program with fellow developers [Discord](https://discord.gg/Kuze7bcT).

**Reply by AbhinavReddy-Dev** [reply]
> Please join this discord server to communicate and pair program with fellow developers [Discord](https://discord.gg/Kuze7bcT).

New link, never expires: https://discord.gg/rVEPt9CK6D

**Comment by AbhinavReddy-Dev** [comment]
Flow for committing changes to main, those who are working on frontend.

1. Check if `frontend` is up to date with `main`.

2. Checkout new `feature_branch` from `frontend`.

3. Develop in `feature_branch`.

4. Create a pull request from `feature_branch` to `frontend` and request for a review before squash and merge.

5. Create a pull request from `frontend` to `main` for rebase and merge.

**Comment by AbhinavReddy-Dev** [comment]
Please create an issue before working on it so that others will know if they have to work on that issue or not, there are a couple cases right now where the teams are working on the issues but have not created an actual issues in the repo, this has created confusions and will make others' efforts undone.

**Comment by abi-t1000** [comment]
@briswells I had couple of questions with your new schema. I am working on charts for summary page. I am planning to create a bar graph for quantity available in each of the four classes that are implementing 'ReportType' interface.
I wanted to know if the quantity is based on each category or is it just the total items.
Like for example, Does 'currentStockReport' has total number of available quantity currently or is it how many quantity available per category ?
